scripts/publish-preview.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const { join } = require('path');
+const glob = require('glob').sync;
+const tiny = require('tiny-json-http');
+const fs = require('fs');
+const { gzipSync } = require('zlib');
+
+async function go() {
+	const { TRAVIS_PULL_REQUEST, TRAVIS_PULL_REQUEST_SHA } = process.env;
+
+	if (!TRAVIS_PULL_REQUEST) { throw new ReferenceError('Missing env var TRAVIS_PULL_REQUEST'); }
+	if (!TRAVIS_PULL_REQUEST_SHA) { throw new ReferenceError('Missing env var TRAVIS_PULL_REQUEST_SHA'); }
+
+	const dir = join(__dirname, '..', 'out');
+	const files = glob(join(dir, '**'), { nodir: true });
+
+	if (!files.length) { throw new ReferenceError('No preview files found to publish'); }
+
+	console.log(`Publishing preview build of PR ${TRAVIS_PULL_REQUEST} (SHA ${TRAVIS_PULL_REQUEST_SHA})`);
+
+	const data = {
+		pr: TRAVIS_PULL_REQUEST,
+		sha: TRAVIS_PULL_REQUEST_SHA,
+		files: [],
+	};
+	for (const file of files) {
+		const filename = file.replace(dir, '').slice(1);
+		const contents = fs.readFileSync(file);
+		const body = gzipSync(contents).toString('base64');
+		console.log(`Packaging: ${filename} (${body.length / 1000}KB)`);
```

Comment:
Nit: the proper notation is what you’d get with `Intl.NumberFormat`, which is “1234 kB”. Once we start using Node.js 14 to run this script, we could even use `Intl.NumberFormat`:

```js
const formatter = new Intl.NumberFormat('en', {
  style: 'unit',
  unit: 'kilobyte',
});
formatter.format(1.234);
// → '1.234 kB'
```

https://v8.dev/features/intl-numberformat#units

Reply:
the colloquial one is "KB", and this is a debugging message, i'm not sure why it's important to be precise here?